### PR TITLE
ci(aio): Refactored payload size script and track hello-world payload

### PR DESCRIPTION
**Pull request by tinayuangao**
…size for hello-world cli project

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[x ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
- Refactored payload-size script so it's re-usable
- Added script in `run_tests.sh` to generate a new hello-world project using angular-cli
- Track all the project size in `integration` if they have a `dist` folder

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by gkalpak — COMMENTED** [review]
`scripts/ci/payload-size.sh` is kind of hard to follow (with all the globals etc), but other than that LGTM (afaict - you can never be sure with bash :smiley:)

File: integration/run_tests.sh [hunk 1]
```diff
@@ -14,6 +17,12 @@ rm_cache
 mkdir $cache
 trap rm_cache EXIT
 
+# Create and build a new Angular project by angular-cli
+testDir="hello_world__cli"
+npm install -g @angular/cli
```
Comment (IgorMinar) [inline integration/run_tests.sh]: I think we should pin this dependency, otherwise we can get easily broken when a cli release with regression happens.

I suggest we add a new global variable into `env.sh` specifying the cli version that should be used for this test, we do the same thing with chrome version or yarn.

**Comment by tinayuangao** [comment]
@IgorMinar Added comments, and checked payload size for `hello_world__cli` and `hello_world__closure`. Will track umd size in a separate pr. PTAL. Thanks for review!

**Comment by tinayuangao** [comment]
Sync and rebased on Misko's PR. Please take another look. Thanks!

**Comment by tinayuangao** [comment]
Added tracking `umd` size part. Please take another look. Thanks! @IgorMinar

File: integration/run_tests.sh [hunk 2]
```diff
@@ -34,7 +38,17 @@ for testDir in $(ls | grep -v node_modules) ; do
     cd $testDir
     # Workaround for https://github.com/yarnpkg/yarn/issues/2256
     rm -f yarn.lock
+    rm -rf dist
     yarn install --cache-folder ../$cache
     yarn test || exit 1
+    # Track payload size for cli-hello-world and hello_world__closure
```
Comment (mhevery) [inline integration/run_tests.sh]: Could we just add `yarn build` before `yarn test` and make sure that all of the subdirectories have an appropriate `yarn build` command even if it is noop?

**Review by mhevery — CHANGES_REQUESTED** [review]
One change, LGTM otherwise

File: integration/run_tests.sh (hunk 2 above)
Comment (tinayuangao) [inline integration/run_tests.sh]: Done. Added `ngc` or `echo noop` in `package.json` files. PTAL. Thanks!

**Review by mhevery — APPROVED** [review]
